src/LinuxCaptureService/LinuxCaptureService.cpp

```diff
@@ -147,6 +149,90 @@ class ProducerEventProcessorHijackingFunctionEntryExitForLinuxTracing
 
 }  // namespace
 
+orbit_capture_service::CaptureService::StopCaptureReason
+LinuxCaptureService::WaitForStopCaptureRequestOrMemoryThresholdExceeded(
+    grpc::ServerReaderWriter<orbit_grpc_protos::CaptureResponse, orbit_grpc_protos::CaptureRequest>*
+        reader_writer) {
+  // wait_for_stop_capture_request_thread_ below outlives this method, hence the shared pointers.
+  auto stop_capture_mutex = std::make_shared<absl::Mutex>();
+  auto stop_capture = std::make_shared<bool>(false);
+  auto stop_capture_reason = std::make_shared<std::optional<StopCaptureReason>>();
+
+  static const std::optional<uint64_t> mem_total_bytes = ReadMemTotalInBytesFromProcMeminfo();
+  std::thread memory_watchdog_thread;
+  if (!mem_total_bytes.has_value()) {
+    ERROR("Could not start memory watchdog because total physical memory is unknown");
+  } else {
+    uint64_t watchdog_threshold_bytes = mem_total_bytes.value() / 2;
+    static constexpr absl::Duration kWatchdogPollInterval = absl::Seconds(1);
+    LOG("Starting memory watchdog with threshold %u B because total physical memory is %u B",
+        watchdog_threshold_bytes, mem_total_bytes.value());
+
+    memory_watchdog_thread = std::thread{
+        [watchdog_threshold_bytes, stop_capture_mutex, stop_capture, stop_capture_reason] {
+          while (true) {
+            {
+              absl::MutexLock lock{&*stop_capture_mutex};
```

Comment:
Super-Nit: `absl::MutexLock lock{stop_capture_mutex.get()}` feels a bit more clear to me.

Reply:
Please ignore if you don't see it that way.